### PR TITLE
Fix process exit code

### LINKED ISSUE #4368: Webdriverio v 5.12.1 return EXIT CODE "0" even after test fails on CI and the job looks like it passed but in real it failed
**Environment:**
 - **WebdriverIO version:5.12.1** [e.g. 4.13.2]
 - **Mode:WDIO Testrunner**
 - **If WDIO Testrunner, running sync/async:sync**
 - **Node.js version:10.13**
 - **NPM version:6.4.1**
 - **Browser name and version:Mobile Native iOS**
 - **Additional wdio packages used (if applicable):@wdio/cucumber-framework** 

**Config of WebdriverIO**
My hooks
```javascript
    /**
     * Runs before a WebdriverIO command gets executed.
     * @param {String} commandName hook command name
     * @param {Array} args arguments that command would receive
     */
    // beforeCommand: function (commandName, args) {
    // },
    /**
     * Runs before a Cucumber feature
     */
    // beforeFeature: function (uri, feature, scenarios) {
    // },
    /**
     * Runs before a Cucumber scenario
     */
    // beforeScenario: function (uri, feature, scenario, sourceLocation) {
    // },
    /**
     * Runs before a Cucumber step
     */
    // beforeStep: function (uri, feature) {
    // },
    /**
     * Runs after a Cucumber step
     */
    // afterStep: function (uri, feature, { error, result }) {
    // },
    /**
     * Runs after a Cucumber scenario
     */
    afterScenario: function (uri, feature, scenario, result, sourceLocation) {
      //Takes Screenshot when test failed
      if (result.status === "failed") {
        browser.takeScreenshot();
      }
    },
    /**
     * Runs after a Cucumber feature
     */
    // afterFeature: function (uri, feature, scenarios) {
    // },

    /**
     * Runs after a WebdriverIO command gets executed
     * @param {String} commandName hook command name
     * @param {Array} args arguments that command would receive
     * @param {Number} result 0 - command success, 1 - command error
     * @param {Object} error error object if any
     */
    // afterCommand: function (commandName, args, result, error) {
    // },
    /**
     * Gets executed after all tests are done. You still have access to all global variables from
     * the test.
     * @param {Number} result 0 - test pass, 1 - test fail
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs List of spec file paths that ran
     */
    // after: function (result, capabilities, specs) {
    // },
    /**
     * Gets executed right after terminating the webdriver session.
     * @param {Object} config wdio configuration object
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs List of spec file paths that ran
     */
    // afterSession: function (config, capabilities, specs) {
    // },
    /**
     * Gets executed after all workers got shut down and the process is about to exit. An error
     * thrown in the onComplete hook will result in the test run failing.
     * @param {Object} exitCode 0 - success, 1 - fail
     * @param {Object} config wdio configuration object
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {<Object>} results object containing test results
     */
    onComplete: function (exitCode, config, capabilities, results) {
      console.log('exit code is ',exitCode);
    },
    /**
     * Gets executed when a refresh happens.
     * @param {String} oldSessionId session ID of the old session
     * @param {String} newSessionId session ID of the new session
     */
    //onReload: function(oldSessionId, newSessionId) {
    //}
```

**Describe the bug**
When running regression on CI machine or in local the test runs fine and prints out the result of how many passed and how many failed but the process always return EXIT_CODE "0" even if the test fails
I used command `echo $?` to print EXIT_CODE and it always prints 0 for webdriverio tests when i upgraded to 5.12.1
I went to `onComplete` hook and printed exitCode on console and it prints "1" but somehow the overall process is returning EXIT_CODE as "0"

**Log**
These are my console logs when i ran a failed test locally 
```
    at process._tickCallback (internal/process/next_tick.js:68:7)
2019-08-15T16:55:21.117Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code 1
[0-0] FAILED in /Users/mer2350/Downloads/nm-app-debug.apk - /test/features/regressionTwo/investments/verifyManualInvestAccountDeleted.feature
2019-08-15T16:55:21.119Z INFO @wdio/cli:Launcher: Run onComplete hook
2019-08-15T16:55:21.119Z INFO @wdio/selenium-standalone-service: shutting down all browsers
2019-08-15T16:55:21.120Z DEBUG @wdio/appium-service: Appium (pid: 98233) killed
exit code is  1

Spec Files:	 0 passed, 1 failed, 1 total (100% completed) in 00:00:33 

2019-08-15T16:55:31.109Z INFO @wdio/local-runner: Shutting down spawned worker
2019-08-15T16:55:31.362Z INFO @wdio/local-runner: Waiting for 0 to shut down gracefully
2019-08-15T16:55:31.362Z INFO @wdio/local-runner: shutting down
MER2350:test-mobile-nmcx mer2350$ echo $?
0
MER2350:test-mobile-nmcx mer2350$ 
```

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

After recent changes wdio cli always exits with code 0.
This PR fixes #4368 and doesn't affect recently fixed "process.exit" / "webdriver session hanging" issues.

- exit process with appropriate exit code after runner.shutdown
- fixed shutdown timeout that causes process to hang
- reverted interface changes made in https://github.com/webdriverio/webdriverio/pull/4332

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/wdio-cli/tests/run.test.js [hunk 1]
```diff
@@ -1,12 +1,53 @@
...
+        }
+        expect(error).toBe('foobar')
+        expect(global.console.error).toBeCalled
+        expect(process.exit).toBeCalledWith(1)
+    })
+
+    afterEach(() => {
+        Launcher.mockClear()
+        global.console.error.mockClear()
+        process.exit.mockClear()
```
Comment (christian-bromann) [inline packages/wdio-cli/tests/run.test.js]: Can we assign the original exit method to it again in the `afterAll` hook?

Comment (mgrybyk) [reply packages/wdio-cli/tests/run.test.js]: Yes, I'll do it later today.

**Comment by nicholasbaileyrally** [comment]
This feels like an issue that needs an integration test of some sort. Given that a non-zero exit code is the standard way of identifying failures for a number of CI platforms, this is a feature we really can't risk casually breaking.

**Comment by mgrybyk** [comment]
We now have unit tests for it

**Comment by catriuspham** [comment]
@mgrybyk Which version will include this change? I'm using webdriverIO version 5.12.4 and CircleCI is still green if the test suite fails.

**Comment by mgrybyk** [comment]
Should be fixed in 5.12.4. Make sure to update wdio packages properly.

**Comment by catriuspham** [comment]
Ah, turned out we also had to upgrade `@wdio/cli` and `@wdio/local-runner`. It works now 👍

**Comment by feryardiant** [comment]
Hi there,

I just got similar issue but with different scenario, the runner stuck [like forever](https://github.com/feryardiant/wpdev/commit/7bda293e37cabdcd3425e9e98ccee95837a6b231/checks?check_suite_id=252341790). I am not sure if it was related but just in case someone else had similar issue like mine, invoking `done()` inside `onComplete` hook, did the trick.

https://github.com/feryardiant/wpdev/blob/135fdfc85ed682992ecc5eee9dc92c2f06c20e7e/gulpfile.js#L240-L254

**Pull request merged**